blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ScheduleContext.java

```diff
@@ -151,9 +151,11 @@ public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards
         this.clock = clock;
         registerMBean();
     }
-
     public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards) {
-        this(currentTimeMillis, managedShards, new DefaultClockImpl());
+        this.scheduleTime = currentTimeMillis;
+        this.shardStateManager = new ShardStateManager(managedShards, asMillisecondsSinceEpochTicker());
+        this.lockManager = new NoOpShardLockManager();
+        this.clock = new DefaultClockImpl();
```

Comment:
My first instinct is to have one and only one constructor that sets fields, and to have all other constructors call that one. However, in this case, java didn't like the call to `asMillisecondsSinceEpochTicker` (i.e. referencing `this`) happening before the call to the supertype constructor is called. Also, it seemed strange to have the other constructors defer to one that was `@VisibleForTesting`. My second choice was to have all of the constructors set fields, and none defer to any other, because that way at least they're consistent. I have changed it to just remove the `registerMBean` call in the second constructor.